x/pairing/keeper/scores/pairing_slot.go

```diff
@@ -0,0 +1,43 @@
+package scores
+
+// PairingSlot holds the set of requirements for a slot.
+// It also holds the number of identical slots required for the pairing (count)
+type PairingSlot struct {
+	Reqs  map[string]ScoreReq
+	Count int
+}
+
+func NewPairingSlot() *PairingSlot {
+	return &PairingSlot{Count: 1}
+}
+
+// Subtract generates a diff slot that contains the reqs that are in the slot receiver but not in the "other" slot
+func (s PairingSlot) Subtract(other *PairingSlot) *PairingSlot {
+	reqsDiff := make(map[string]ScoreReq)
+	for key := range s.Reqs {
+		if _, found := other.Reqs[key]; !found {
```

Comment:
fetch the req here instead of fetching it 2 lines below again

Reply:
not what i meant,
otherValue , found := other.Reqs[key]
than you can use it in the else if

Reply:
done. see https://github.com/lavanet/lava/commit/6ba55d5846fd3f875247146f357d52c5249e7d4a